docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,468 @@
+    {
+        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        // Setting TCP Keep Alive
+        socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
+        socket.SetSocketOption(SocketOptionLevel.Tcp, SocketOptionName.TcpKeepAliveTime, 60);
+        socket.SetSocketOption(SocketOptionLevel.Tcp, SocketOptionName.TcpKeepAliveInterval, 1);
+
+        // Setting ReceiveBufferSize
+        socket.ReceiveBufferSize = 8192;
+
+        // Enabling ReusePort
+        socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseUnicastPort, true);
+
+        // Disabling Nagle Algorithm
+        socket.NoDelay = true;
+
+        await socket.ConnectAsync(context.DnsEndPoint, cancellationToken);
+        return new NetworkStream(socket, ownsSocket: true);
+    }
+};
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+```
+
+### Example: Enabling Dns Round Robin
+
+DNS Round Robin is a technique used to distribute network traffic across multiple servers by rotating through a list of IP addresses associated with a single domain name. This helps in load balancing and improving the availability of services. When using HttpClient, you can implement DNS Round Robin by manually handling the DNS resolution and rotating through the IP addresses using the ConnectCallback property of SocketsHttpHandler.
+
+By enabling DNS Round Robin, you can ensure that your application distributes requests evenly across multiple servers, reducing the load on any single server and improving overall performance and reliability. This approach is particularly useful for high-traffic applications that require efficient load balancing.
+
+To enable DNS Round Robin with HttpClient, you can use the ConnectCallback property to manually resolve the DNS entries and rotate through the IP addresses. Here's an example for `HttpWebRequest` and `HttpClient`:
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+ServicePointManager.DnsRefreshTimeout = 60000;
+ServicePointManager.EnableDnsRoundRobin = true;
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+```
+
+**New Code Using `HttpClient`**:
+
+```csharp
+ConcurrentDictionary<string, DnsCache> dnsCache = new(StringComparer.OrdinalIgnoreCase);
+
+TimeSpan dnsRefreshTimeout = TimeSpan.FromMinutes(1);
+var handler = new SocketsHttpHandler
+{
+    ConnectCallback = async (context, cancellationToken) =>
+    {
+        string host = context.DnsEndPoint.Host;
+        
+        DnsCache? cacheItem;
+        if (!dnsCache.TryGetValue(host, out cacheItem) || DateTime.Now > cacheItem.CacheExpireTime)
+        {
+            dnsCache.TryRemove(host, out _);
+            cacheItem = new DnsCache(await Dns.GetHostAddressesAsync(host, cancellationToken), DateTime.Now.Add(dnsRefreshTimeout), 0);
+            dnsCache.TryAdd(host, cacheItem);
```

Comment:
`ConnectCallback` doesn't guarantee thread safety. This will work, but it might unnecessarily issue multiple DNS requests for the same host, while also unintentionally removing freshly inserted results.